### ISSUE TITLE
Enums with dots in their names cause an error

### ISSUE THREAD

**Issue by fr3fou** [anymaniax/orval#160]
## What are the steps to reproduce this issue?
1. Make a swagger schema with an enum that contains a dot in the name `foo.bar`
2. Use it inside a definition

```json
{
   "web.foo": {
            "type": "object",
            "properties": {
                "id": {
                    "type": "integer"
                },
                "name": {
                    "type": "string"
                },
                "foo": {
                    "type": "string",
                    "enum": [
                        "high",
                        "medium",
                        "low"
                    ]
                }
            }
        }
}
```

## What happens?
The generated code is wrong:
![image](https://user-images.githubusercontent.com/1344906/120635653-63d42f80-c475-11eb-9c96-5c0fd9039794.png)

## What were you expecting to happen?
The dot to be removed

## What versions are you using?
**Operating System:** macOS Big Sur
**Package Version:** 5.4.4

**Comment by Eddie-CooRo** [comment]
I've also faced a somewhat similar issue when trying to use a string enum with some of the values being unusable in ts keys.
```json
{
  "type": "object",
  "properties": {
    "sortOrder": {
      "type": "string",
      "enum": ["-1", "0", "1"],
    },
  },
}
```
The generated output will be:

![image](https://user-images.githubusercontent.com/20463083/120677610-80d82500-c4ac-11eb-8aba-f3ec5ddd59bf.png)

With the duplicated declaration of `"1"`

**Comment by fr3fou** [comment]
Is there any temporary workaround? I can't really change my schema as it's autogenerated

**Comment by anymaniax** [comment]
I will take time on it today and I don't think there is a workaround

**Comment by anymaniax** [comment]
@Eddie-CooRo I did something like this it seems ok for you?
<img width="521" alt="Screenshot 2021-06-04 at 15 32 31" src="https://user-images.githubusercontent.com/10516382/120809324-386f4480-c54a-11eb-8b8a-59efb263d5da.png">

**Comment by Eddie-CooRo** [comment]
YES! That's more than enough for me.
Thanks :pray: 

Though the most ideal solution would be for open API to add support for named enums :)
```
{
  type: "string",
  enum: [{ name: "Asc", value: 1 }, { name: "Desc", value: -1 }, { name: "None", value: 0 }],
}
```
And
```
export const SortOrder = {
  Asc: 1,
  Desc: -1,
  None: 0
}
```

**Comment by anymaniax** [comment]
I fixed the problem in the `5.4.6` normally

**Comment by fr3fou** [comment]
Thank you so much! Great job with this tool, been loving it!
